org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionScanner.java

```diff
@@ -904,7 +904,7 @@ protected int getNextNotFakedToken() throws InvalidInputException {
 protected int scanForTextBlock() throws InvalidInputException {
 	int lastQuotePos = 0;
```

Comment:
Since this method overrides the super method completely and the super method has various changes specifically for templates, it is likely that completion inside text blocks is broken. We need tests to establish what works and what doesn't and raise follow up tickets

Reply:
Completion after a template works after a minor fix. Completion inside a template expression doesn't. I have raised #1601 for this.